docs/rule/require-has-block-helper.md

```diff
@@ -0,0 +1,54 @@
+# require-has-block-helper
+
+In Ember 3.25 the properties `hasBlock` and `hasBlockParams` were deprecated. Their replacement is to use `has-block` and `has-block-params` helpers instead.
+
+This rule prevents the usage of `hasBlock` and `hasBlockParams` and suggests using `has-block` or `has-block-params` instead.
+
+For more information about this deprecation you can view the [RFC](https://github.com/emberjs/rfcs/blob/master/text/0689-deprecate-has-block.md) or its entry on the [Deprecations page](https://deprecations.emberjs.com/v3.x/#toc_has-block-and-has-block-params).
+## Examples
+
+This rule **forbids** the following:
+
+```hbs
+{{hasBlock}}
+{{#if hasBlock}}
+
+{{/if}}
+```
+
+```hbs
+{{hasBlockParams}}
+{{#if hasBlockParams}}
+
+{{/if}}
+```
+
+This rule **allows** the following:
+
+```hbs
+{{has-block}}
```

Comment:
is it make sence to aloow plain `{{has-block}}` usage? (I suppose it should be only inside expressions)
cc @rwjblue

Reply:
same for `has-block-params`, and wondering, should we require block name  `(has-block "default")` ? (imho - yes)

Reply:
So according to the [deprecation guide](https://github.com/emberjs/rfcs/blob/master/text/0689-deprecate-has-block.md#deprecation-guide) in the RFC `{{has-block}}` and `{{has-block-params}}` are both valid.

> You may optionally pass a name to (has-block), the name of the block to check. The name corresponding to the block that {{hasBlock}} represents is "default". Calling (has-block) without any arguments is equivalent to calling (has-block "default").

Reply:
@patocallaghan, yes, but i'm trying to figure out real use-cases for exact usage of `{{has-block}}`, `{{has-block-params}}` (outside expressions), and for this 1% I suppose easier to disable rule.

regarding requred `default` - ember trying to be more explicit and straightforward, in `3.25` we have named blocks enabled by default, and think it's good approach to requre block name in this helpers.

Reply:
Sure fair point.

@rwjblue I know the RFC called out supporting not passing "default" but do you have any thoughts should we be recommending the use of `{{has-block "default'}}`? If that is the case, should the fixer I'm going to work on migrate things to use the expanded `{{has-block "default"}}` syntax rather than just `{{has-block}}`?

Reply:
> If that is the case, should the fixer I'm going to work on migrate things to use the expanded {{has-block "default"}} syntax rather than just {{has-block}}?

The fixer should use whatever the `hasBlock`/`hasBlockParams` call was passed (so if it was passing a positional param string, the fixer should preserve that).

Reply:
>  should we require block name (has-block "default") ? (imho - yes)

I don't think we should. 

See https://github.com/ember-template-lint/ember-template-lint/issues/1352.